src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -79,6 +100,21 @@ public class JsonMessageFormatter : IJsonRpcMessageTextFormatter
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private Encoding encoding;
 
+        /// <summary>
+        /// Incrementable number to assing as token for the progress objects.
+        /// </summary>
+        private long nextProgressId;
+
+        /// <summary>
+        /// Stores the request ID so that the converter can use it to create the request-progress map.
+        /// </summary>
+        private long? requestId;
```

Comment:
>requestId [](start = 22, length = 9)

Isn't this class used for all incoming requests?  Why keep a specific request ID as a field in the class?

Reply:
The xml doc comment can probably use some beefing up. It should say something like 

> Stores the id of the request currently being serialized so the converter can use it to create the request-progress map.

Also: let's rename it to `requestIdBeingSerialized` or something like that.

---
In reply to: [304598499](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r304598499) [](ancestors = 304598499)